api/envoy/api/v2/lds.proto

```diff
@@ -238,4 +238,11 @@ message Listener {
   // If no configuration is specified, Envoy will not attempt to balance active connections between
   // worker threads.
   ConnectionBalanceConfig connection_balance_config = 20;
+
+  // When this flag is set to true, listeners sets *SO_REUSEPORT* socket option and
+  // create one socket for each worker thread, this makes inbound connections
+  // distributed among worker threads evenly. When this flag is set to false,
+  // all worker threads shares one socket.
+  // For UDP this flag is set to true forcibly.
+  google.protobuf.BoolValue reuse_port = 21;
```

Comment:
This can probably be just a `bool`, as the default is false (even with the UDP override).

Reply:
+1

Here is edited text that I would use:
```
 // When this flag is set to true, listeners set the *SO_REUSEPORT* socket option and
 // create one socket for each worker thread. This makes inbound connections
 // distribute among worker threads roughly evenly in cases where there are a high number
 // of connections from different sources to hash. When this flag is set to false,
 // all worker threads share one socket. For UDP this flag is set to true forcibly.
```

Also, please add a release note with a ref link to this field.

Also, is it still true that when using SO_REUSEPORT for TCP that connections can sometimes be dropped? I know that used to be the case. If so, should we warn about that in hot restart cases?